### ISSUE TITLE
build.py Exception: CalledProcessError

### ISSUE THREAD

**Issue by Peterscraps** [Silverlan/pragma#32]
For some reason `os.getcwd` sets the terminal directory to the root directory+/pragma instead of getting the current directory (would os.curdir work instead?)
![image](https://user-images.githubusercontent.com/101078225/216159730-462ef4cc-e776-4970-8f08-cc651eb2eaa1.png)
where I would expect c:/pragma/pragma/pragma (the same directory the terminal was run from)
And yea it is a little silly, iv'e gained another pragma subfolder, an oopsy of mine and probably the cause. Problem is I can't move it with git attached to the dependencies folder. 

Run without debugging using --update:
`C:\pragma\pragma\pragma\build\external_libs\util_formatted_text\RelWithDebInfo\util_formatted_text.lib
  wgui.vcxproj -> C:\pragma\pragma\pragma\build\external_libs\wgui\RelWithDebInfo\wgui.dll
  servermanager.vcxproj -> C:\pragma\pragma\pragma\build\external_libs\networkmanager\RelWithDebInfo\servermanager.lib
Traceback (most recent call last):
  File "C:\pragma\pragma\pragma\build_scripts\build.py", line 813, in <module>
    cmake_build(build_config,targets)
  File "C:\pragma\pragma\pragma\build_scripts\build.py", line 209, in cmake_build
    subprocess.run(args,check=True)
  File "C:\Users\petes beast\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'RelWithDebInfo', '--target', 'pragma-install-full', 'pr_prosper_vulkan', 'pr_bullet', 'pr_audio_soloud', 'pr_curl', 'pr_dmx', 'pr_chromium', 'pr_unirender', 'pr_curl', 'pr_dmx', 'pr_xatlas', 'pr_openvr', 'pr_prosper_vulkan', 'pr_bullet', 'pr_audio_soloud', 'pr_curl', 'pr_dmx', 'pr_chromium', 'pr_unirender', 'pr_curl', 'pr_dmx', 'pr_xatlas', 'pr_openvr', 'pfm', 'render_raytracing', 'render_raytracing', 'pragma-install']' returned non-zero exit status 1.`

Run with debugging:
`Exception has occurred: CalledProcessError       (note: full exception trace is shown but execution is paused at: <module>)
Command '['cmake', 'C:/pragma/third_party_libs/luajit', '-G', 'Visual Studio 17 2022']' returned non-zero exit status 1.
  File "C:\pragma\pragma\pragma\build_scripts\build.py", line 202, in cmake_configure
    subprocess.run(args,check=True)
  File "C:\pragma\pragma\pragma\build_scripts\build.py", line 421, in <module> (Current frame)
    cmake_configure(root +"/third_party_libs/luajit",generator)
subprocess.CalledProcessError: Command '['cmake', 'C:/pragma/third_party_libs/luajit', '-G', 'Visual Studio 17 2022']' returned non-zero exit status 1.`

**Comment by Silverlan** [comment]
Which terminal is this? Are you sure you're in "c:/pragma/pragma/pragma" when running the script? What does it say if you just enter "cd" (without arguments)?

**Comment by Peterscraps** [comment]
This is a terminal run from VS code, it was ran with the terminal in the directory "c/:pragma/pragma/pragma" for some reason I omitted that that only occurred when run in debugging mode since it halted execution at `Exception has occurred: CalledProcessError       (note: full exception trace is shown but execution is paused at: <module>) Command '['cmake', 'C:/pragma/third_party_libs/luajit', '-G', 'Visual Studio 17 2022']' returned non-zero exit status 1.`
when run normally the terminal would end up back in "c/:pragma/pragma/pragma" I ran the debug today but got different results. I also tried it as a separate admin terminal in case it was to do with permissions but it still exits with the same error as the first error in the OP. 
Iv'e attached a video to clear up any unintended ambiguity.
https://youtu.be/3d_rbil7lJA
